documentation/docs/30-advanced/20-hooks.md

```diff
@@ -232,6 +233,36 @@ During development, if an error occurs because of a syntax error in your Svelte
 
 > Make sure that `handleError` _never_ throws an error
 
+## Isomorphic hooks
+
+The following can be added to `src/hooks.js`. Isomorphic hooks share the same implementation on the server and the client. This is not to be confused with shared hooks, which use separate implementations. 
+
+### rewriteUrl
+
+This function allows you to rewrite URLs before SvelteKit uses them to determine which route to render. It will not change the URL in the browser's address bar, or the value of `event.url`. Only the route matching logic is affected.
+
+```js
+/// file: src/hooks.router.js
+// @errors: 2345
+// @errors: 2304
+
+/** @type {Record<string, string>} */
+const mapping = {
+	"/en/about": "/en/about",
+	"/de/ueber-uns": "/de/about",
+	"/fr/a-propos": "/fr/about",
+}
+
+/** @type {import('@sveltejs/kit').RewriteUrl} */
+export function rewriteUrl({ url }) {
+	url.pathname = mapping[url.pathname] ?? url.pathname;
+	return url;
+}
+```
+
+> `rewriteUrl` is not applied recursively
```

Comment:
it took me a moment to understand what was meant by "recursively" as I was thinking maybe it was referring to deeper paths or something
```suggestion
> `rewriteUrl` is not applied to the returned URL
```